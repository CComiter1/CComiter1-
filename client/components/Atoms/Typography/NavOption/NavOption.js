import React from 'react'
import styled from 'styled-components'

export const Option = styled.a`
  font-family: ${props => props.theme.bodyFontFamily};
  cursor: pointer;
  margin: 0;
  color: white;
  text-decoration: none;
  padding: 1rem 1rem;
  background-color: ${prop => prop.theme.menuColor};
  width: 100%;

  &:hover {
    background-color: ${prop => prop.theme.menuActiveColor};
  }
`

export default (props) => {
  const OptionWrapper = props.active ?
    Option.extend`
      background-color: ${prop => prop.theme.menuActiveColor};
    ` :
    Option
  return (
    <OptionWrapper {...props}>
      {props.children}
    </OptionWrapper>
  )
}

