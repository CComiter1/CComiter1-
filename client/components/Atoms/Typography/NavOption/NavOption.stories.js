import React from 'react'
import { storiesOf } from '@storybook/react'

import NavOption from './NavOption'

storiesOf('Atoms/Typography/NavOption', module)
  .add('inactive', () => <NavOption>Home</NavOption>)
  .add('active', () => <NavOption active>Active</NavOption>)
