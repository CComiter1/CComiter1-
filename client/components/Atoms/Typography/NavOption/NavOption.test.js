import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import NavOption from './NavOption'

describe('NavOption', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <NavOption>ONE OPTION</NavOption>
    ).toJSON()
    expect(component).toMatchSnapshot()
  })
})
