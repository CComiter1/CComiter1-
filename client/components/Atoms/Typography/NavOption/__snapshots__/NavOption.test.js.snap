// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavOption renders correctly 1`] = `
.c0 {
  cursor: pointer;
  margin: 0;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 1rem 1rem;
  width: 100%;
  box-sizing: border-box;
}

<a
  className="c0"
>
  ONE OPTION
</a>
`;
